[doc] Multivalued properties do not accept | as a separator

**Affects PMD Version:** 
7.0.0-rc4

**Description:**
After upgrading from 7.0.0-rc3, the analysis logs exceptions. This appears to only be noise as it does not fail the run.

**Exception Stacktrace:**

```console
Exception applying rule LooseCoupling on file /home/runner/work/caffeine/caffeine/caffeine/src/javaPoet/java/com/github/benmanes/caffeine/cache/Feature.java, continuing with next rule
org.apache.commons.lang3.exception.ContextedRuntimeException: java.lang.IllegalArgumentException: Not a Java binary name 'java.util.Properties         |java.util.concurrent.ConcurrentHashMap         |com.google.common.collect.ImmutableMap         |com.google.common.collect.ImmutableSet         |com.google.common.collect.ImmutableList         |com.github.benmanes.caffeine.cache.BoundedLocalCache         |com.github.benmanes.caffeine.cache.UnboundedLocalCache'
Exception Context:
	[1:Rule applied on node=!debug only! [ClassOrInterfaceType:52:24]ImmutableSet<Feature>]
---------------------------------
	at net.sourceforge.pmd.util.AssertionUtil.contexted(AssertionUtil.java:232)
	at net.sourceforge.pmd.lang.rule.internal.RuleApplicator.applyOnIndex(RuleApplicator.java:77)
	at net.sourceforge.pmd.lang.rule.internal.RuleApplicator.apply(RuleApplicator.java:55)
	at net.sourceforge.pmd.RuleSets.apply(RuleSets.java:158)
	at net.sourceforge.pmd.lang.impl.PmdRunnable.processSource(PmdRunnable.java:140)
	at net.sourceforge.pmd.lang.impl.PmdRunnable.run(PmdRunnable.java:80)
	at java.base@20.0.2/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:577)
	at java.base@20.0.2/java.util.concurrent.FutureTask.run(FutureTask.java:317)
	at java.base@20.0.2/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
	at java.base@20.0.2/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
	at java.base@20.0.2/java.lang.Thread.run(Thread.java:1623)
Caused by: java.lang.IllegalArgumentException: Not a Java binary name 'java.util.Properties         |java.util.concurrent.ConcurrentHashMap         |com.google.common.collect.ImmutableMap         |com.google.common.collect.ImmutableSet         |com.google.common.collect.ImmutableList         |com.github.benmanes.caffeine.cache.BoundedLocalCache         |com.github.benmanes.caffeine.cache.UnboundedLocalCache'
	at net.sourceforge.pmd.util.AssertionUtil.assertValidJavaBinaryName(AssertionUtil.java:55)
	at net.sourceforge.pmd.lang.java.types.TypeTestUtil.isA(TypeTestUtil.java:116)
	at net.sourceforge.pmd.lang.java.rule.bestpractices.LooseCouplingRule.isAllowedType(LooseCouplingRule.java:72)
	at net.sourceforge.pmd.lang.java.rule.bestpractices.LooseCouplingRule.visit(LooseCouplingRule.java:49)
	at net.sourceforge.pmd.lang.java.ast.ASTClassOrInterfaceType.acceptVisitor(ASTClassOrInterfaceType.java:159)
	at net.sourceforge.pmd.lang.java.ast.AbstractJavaNode.acceptVisitor(AbstractJavaNode.java:38)
	at net.sourceforge.pmd.lang.java.rule.AbstractJavaRule.apply(AbstractJavaRule.java:24)
	at net.sourceforge.pmd.lang.rule.AbstractDelegateRule.apply(AbstractDelegateRule.java:232)
	at net.sourceforge.pmd.lang.rule.internal.RuleApplicator.applyOnIndex(RuleApplicator.java:75)
	... 9 more
```
https://github.com/ben-manes/caffeine/actions/runs/6389804412/job/17341828673?pr=1229#step:5:284

**Code Sample demonstrating the issue:**

Project's [rulesSets.xml](https://github.com/ben-manes/caffeine/blob/94a58c4e6b05761efbd5753342dd7a7f95a72df2/gradle/config/pmd/rulesSets.xml#L21-L33).

**Steps to reproduce:**
Check out the PR commit and run the script,
```sh
.github/scripts/analyze.sh
```

**Running PMD through:** *Gradle*

--- a/docs/pages/pmd/userdocs/configuring_rules.md
+++ b/docs/pages/pmd/userdocs/configuring_rules.md
@@ -4,7 +4,7 @@ short_title: Configuring rules
 keywords: [property, properties, message, priority]
 tags: [userdocs, getting_started]
 summary: "Learn how to configure your rules directly from the ruleset XML."
-last_updated: May 2023 (7.0.0)
+last_updated: February 2024 (7.0.0)
 permalink: pmd_userdocs_configuring_rules.html
 author: Hooper Bloob <hooperbloob@users.sourceforge.net>, Romain Pelisse <rpelisse@users.sourceforge.net>, Clément Fournier <clement.fournier76@gmail.com>
 ---
@@ -43,7 +43,10 @@ will cause the rule to be ignored.
 
 ## Rule properties
 
-Properties make it easy to customise the behaviour of a rule directly from the xml. They come in several types, which correspond to the type of their values. For example, NPathComplexity declares a property "reportLevel", with an integer value type, and which corresponds to the threshold above which a method will be reported. If you believe that its default value of 200 is too high, you could lower it to e.g. 150 in the following way:
+Properties make it easy to customise the behaviour of a rule directly from the xml. They come in several types,
+which correspond to the type of their values. For example, NPathComplexity declares a property "reportLevel",
+with an integer value type, and which corresponds to the threshold above which a method will be reported.
+If you believe that its default value of 200 is too high, you could lower it to e.g. 150 in the following way:
 
 ```xml
 <rule ref="category/java/design.xml/NPathComplexity">
@@ -55,7 +58,9 @@ Properties make it easy to customise the behaviour of a rule directly from the x
 </rule>
 ```
 
-Properties are assigned a value with a `property` element, which should mention the name of a property as an attribute. The value of the property can be specified either in the content of the element, like above, or in the `value` attribute, e.g.
+Properties are assigned a value with a `property` element, which should mention the name of a property as an
+attribute. The value of the property can be specified either in the content of the element, like above, or
+in the `value` attribute, e.g.
 
 ```xml
 <property name="reportLevel" value="150"/>
@@ -63,12 +68,17 @@ Properties are assigned a value with a `property` element, which should mention 
 
 All property assignments must be enclosed in a `properties` element, which is itself inside a `rule` element.
 
-{%include tip.html content="The properties of a rule are documented with the rule, e.g. [here](pmd_rules_java_design.html#npathcomplexity) for NPathComplexity. Note that **assigning a value to a property that does not exist throws an error!**" %}
+{% capture tip_content %}
+The properties of a rule are documented with the rule, e.g. [here](pmd_rules_java_design.html#npathcomplexity)
+for NPathComplexity. Note that **assigning a value to a property that does not exist throws an error!**
+{% endcapture %}
+{%include tip.html content=tip_content %}
 
-Some properties take multiple values (a list), in which case you can provide them all by delimiting them with a delimiter character. It is usually a pipe ('\|'), or a comma (',') for numeric properties, e.g.
+Some properties take multiple values (a list), in which case you can provide them all by delimiting them with
+a comma (','), e.g.
 ```xml
 <property name="legalCollectionTypes"
-          value="java.util.ArrayList|java.util.Vector|java.util.HashMap"/>
+          value="java.util.ArrayList,java.util.Vector,java.util.HashMap"/>
 ```
 
 These properties are referred to as **multivalued properties** in this documentation.

--- a/docs/pages/release_notes.md
+++ b/docs/pages/release_notes.md
@@ -255,6 +255,7 @@ The rules have been moved into categories with PMD 6.
   * [#4659](https://github.com/pmd/pmd/pull/4659):   \[doc] Improve ant documentation
   * [#4669](https://github.com/pmd/pmd/pull/4669):   \[doc] Add bld PMD Extension to Tools / Integrations
   * [#4676](https://github.com/pmd/pmd/issues/4676): \[doc] Clarify how CPD `--ignore-literals` and `--ignore-identifiers` work
+  * [#4704](https://github.com/pmd/pmd/issues/4704): \[doc] Multivalued properties do not accept | as a separator
 * miscellaneous
   * [#4699](https://github.com/pmd/pmd/pull/4699):   Make PMD buildable with java 21
   * [#4586](https://github.com/pmd/pmd/pull/4586):   Use explicit encoding in ruleset xml files
@@ -1305,6 +1306,7 @@ See also [Detailed Release Notes for PMD 7]({{ baseurl }}pmd_release_notes_pmd7.
     * [#4676](https://github.com/pmd/pmd/issues/4676): \[doc] Clarify how CPD `--ignore-literals` and `--ignore-identifiers` work
     * [#4659](https://github.com/pmd/pmd/pull/4659):   \[doc] Improve ant documentation
     * [#4669](https://github.com/pmd/pmd/pull/4669):   \[doc] Add bld PMD Extension to Tools / Integrations
+    * [#4704](https://github.com/pmd/pmd/issues/4704): \[doc] Multivalued properties do not accept | as a separator
 * testing
     * [#2435](https://github.com/pmd/pmd/issues/2435): \[test] Remove duplicated Dummy language module
     * [#4234](https://github.com/pmd/pmd/issues/4234): \[test] Tests that change the logging level do not work

